src/parser/scanner.ll

```diff
@@ -208,6 +209,7 @@ WS [ \t\r\v\f]
 "to_string"                           { return yy::parser::make_TOSTRING(yylloc); }
 "to_unsigned"                         { return yy::parser::make_TOUNSIGNED(yylloc); }
 "choice-domain"                       { return yy::parser::make_CHOICEDOMAIN(yylloc); }
+"__ITERATION__"                       { return yy::parser::make_ITERATION(yylloc); }
```

Comment:
Should the iteration counter be made available as an intrinsic functor form like the `autoinc()` ?

Reply:
Yes, I wasn't sure how to name it. Something like "main_loop_iteration()" ?

Reply:
I'm fine with that, the identifier looks long enough to not clash with existing code.